asllib/Typing.ml

```diff
@@ -1247,7 +1247,7 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
           (Serialize.subprogram_type_to_string call_type)
           PP.pp_pos loc
     in
-    let caller_arg_typed = List.map (annotate_expr env) args in
+    let caller_arg_typed = List.map (annotate_expr_def env) args in
```

Comment:
Hi @HadrienRenaud. This PR changes code to ignore some warnings (48,  [eliminated-optional-arguments]: implicit elimination of optional argument ?forbid_atcs).

This fix consists in adding some argument for the optional argument to take its default value. See below for the definition of `annotate_expr_def`.
```OCaml
and annotate_expr_def env e = annotate_expr env e
```
In some occasions for instance in `annotate_call`, this choice by default may not be the right one. Alternatively, one could have added `?forbid_atcs` as an argument to `annotate_call` for it to be passed to the recursive call to `annotate_expr` on the arguments of the call. Your opinion?